### ISSUE TITLE
🐛 BUG: `wrangler publish` works but throws `workers.api.error.script_not_found [code: 10007]`

### ISSUE THREAD

**Issue by Keavon** [cloudflare/workers-sdk#2218]
### What version of `Wrangler` are you using?

2.3.2

### What operating system are you using?

Windows with cmd.exe

### Describe the Bug

I'm creating my first Worker from Wrangler. I started by cloning an existing worker with `wrangler init --from-dash` that I had written in the web UI, now I've turned it into TypeScript split across multiple files and renamed it in the web UI as well as wrangler.toml.

Now when I run `wrangler publish` I'm getting this error in the console: `workers.api.error.script_not_found [code: 10007]` and in the web UI I get spammed with `We could not find that Workers Service on your account. (Code: 10007)` toast notifications. However, I can click the web UI's edit button and view the script which was successfully updated from my local changes via Wrangler.

Logged in via OAuth.

**Comment by JacobMGEvans** [comment]
Greetings 👋🏻, 

What is the script name that you are trying to publish, does it match the name in Wrangler config (`wrangler.toml`) or the name in command `wrangler publish [SCRIPT] --name [SCRIPT_NAME]`

Cc: @rozenmd

**Comment by Keavon** [comment]
I confirmed the script name is correct in `wrangler.toml` but I also ran it again with `wrangler publish [SCRIPT]`. In the command you gave, what's the difference between `[SCRIPT]` and `--name [SCRIPT_NAME]`?

In my monorepo, this is in `/infrastructure/editor-api`. That directory contains `wrangler.toml`, `tsconfig.json`, `package.json`, `package-lock.json`, `node_modules/`, and `src/`.

`src/` has `main.ts` (my entry point) and some other TS files as siblings and as children of another folder.

My wrangler.toml starts with:

```toml
name = "graphite-editor-api"
main = "src/main.ts"
compatibility_date = "2022-11-15"
route = "api.graphite.rs/font-list"
usage_model = "bundled"
env = { }
```

`name` is the updated name for the worker after I renamed it in the dashboard and in this line of the file
`route` is unchanged from before I renamed it, since that aspect didn't change

I am running from the `cmd.exe` working directory of `[monorepo]\infrastructure\editor-api`.
